appservice/src/createAppService/createAppService.ts

```diff
@@ -24,13 +26,67 @@ export async function createAppService(
     credentials: ServiceClientCredentials,
     subscriptionId: string,
     subscriptionDisplayName: string,
-    showCreatingNode?: (label: string) => void): Promise<Site> {
+    showCreatingNode?: (label: string) => void,
+    fsPath?: string,
+    improvedCreation: boolean = false): Promise<Site> {
+
+    let wizardContext: IAppServiceWizardContext = {
+        newSiteKind: appKind,
+        newSiteOS: websiteOS,
+        subscriptionId: subscriptionId,
+        subscriptionDisplayName: subscriptionDisplayName,
+        credentials: credentials
+    };
 
     const promptSteps: AzureWizardPromptStep<IAppServiceWizardContext>[] = [];
     promptSteps.push(new SiteNameStep());
-    promptSteps.push(new ResourceGroupListStep());
+    promptSteps.push(new LocationListStep());
     promptSteps.push(new SiteOSStep());
     promptSteps.push(new SiteRuntimeStep());
+    if (improvedCreation) {
+        // function used to set name property after location and OS has been selected
+        const setResourceGroupProperty: Function = (): string => {
```

Comment:
TODO

Reply:
The functions below are passed into the SetWizardPropertySteps to be executed later to set the rg and asp names which this comment is referring to.  I rewrote it to be more clarifying.